aiokafka/protocol/commit.py

```diff
@@ -244,69 +263,36 @@ class OffsetFetchRequest_v2(Request):
     SCHEMA = OffsetFetchRequest_v1.SCHEMA
 
 
-class OffsetFetchRequest_v3(Request):
+class OffsetFetchRequest_v3(RequestStruct):
     API_KEY = 9
     API_VERSION = 3
     RESPONSE_TYPE = OffsetFetchResponse_v3
     SCHEMA = OffsetFetchRequest_v2.SCHEMA
 
 
-OffsetFetchRequest = [
-    OffsetFetchRequest_v0,
-    OffsetFetchRequest_v1,
-    OffsetFetchRequest_v2,
-    OffsetFetchRequest_v3,
-]
-OffsetFetchResponse = [
-    OffsetFetchResponse_v0,
-    OffsetFetchResponse_v1,
-    OffsetFetchResponse_v2,
-    OffsetFetchResponse_v3,
-]
-
-
-class GroupCoordinatorResponse_v0(Response):
-    API_KEY = 10
-    API_VERSION = 0
-    SCHEMA = Schema(
-        ("error_code", Int16),
-        ("coordinator_id", Int32),
-        ("host", String("utf-8")),
-        ("port", Int32),
-    )
-
-
-class GroupCoordinatorResponse_v1(Response):
-    API_KEY = 10
-    API_VERSION = 1
-    SCHEMA = Schema(
-        ("throttle_time_ms", Int32),
-        ("error_code", Int16),
-        ("error_message", String("utf-8")),
-        ("coordinator_id", Int32),
-        ("host", String("utf-8")),
-        ("port", Int32),
-    )
-
-
-class GroupCoordinatorRequest_v0(Request):
-    API_KEY = 10
-    API_VERSION = 0
-    RESPONSE_TYPE = GroupCoordinatorResponse_v0
-    SCHEMA = Schema(
-        ("consumer_group", String("utf-8")),
-    )
-
-
-class GroupCoordinatorRequest_v1(Request):
-    API_KEY = 10
-    API_VERSION = 1
-    RESPONSE_TYPE = GroupCoordinatorResponse_v1
-    SCHEMA = Schema(
-        ("coordinator_key", String("utf-8")),
-        ("coordinator_type", Int8),
-    )
-
-
-GroupCoordinatorRequest = [GroupCoordinatorRequest_v0, GroupCoordinatorRequest_v1]
-GroupCoordinatorResponse = [GroupCoordinatorResponse_v0, GroupCoordinatorResponse_v1]
+class OffsetFetchRequest(Request):
+    API_KEY = 9
+    CLASSES = [
+        OffsetFetchRequest_v1,
+        OffsetFetchRequest_v2,
+        OffsetFetchRequest_v3,
+    ]
+
+    def __init__(
+        self, consumer_group: str, partitions: list[tuple[str, list[int]]] | None
+    ):
+        self._consumer_group = consumer_group
+        self._partitions = partitions
+
+    def build(self, request_struct_class: type[RequestStruct]) -> RequestStruct:
+        if request_struct_class.API_VERSION < 2 and self._partitions is None:
+            raise IncompatibleBrokerVersion(
```

Comment:
Looks like the exception changed for this case. This one looks more appropriate to me, but I'm not sure if we don't need to mention about it in the change log.